### PR TITLE
Add enrich policy execute API

### PR DESCRIPTION
**Pull request by hub-cap**
This commit wires up the Rest calls and Transport calls for execute
enrich policy, as well as tests and rest spec additions.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/ExecuteEnrichPolicyAction.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public static final ExecuteEnrichPolicyAction INSTANCE = new ExecuteEnrichPolicyAction();
+    public static final String NAME = "cluster:admin/xpack/enrich/execute";
+
+    private ExecuteEnrichPolicyAction() {
+        super(NAME);
+    }
+
+    @Override
+    public AcknowledgedResponse newResponse() {
+        return new AcknowledgedResponse();
```
Comment (martijnvg) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/ExecuteEnrichPolicyAction.java]: Maybe we should define a different response class here? Right now we return only a boolean, but we return more in the future and then it is easier bwc wise if we have a custom class? Also toXContent() returns `acknowledge` in the body and I think something like `executed` is better?

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/rest/RestExecuteEnrichPolicyAction.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.enrich.action.ExecuteEnrichPolicyAction;
+
+import java.io.IOException;
+
+public class RestExecuteEnrichPolicyAction extends BaseRestHandler {
+
+    public RestExecuteEnrichPolicyAction(final Settings settings, final RestController controller) {
+        super(settings);
+        controller.registerHandler(RestRequest.Method.PUT, "/_enrich/policy/{name}/_execute", this);
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/rest/RestExecuteEnrichPolicyAction.java]: I think we should add a POST variant here too. We do that for other execute APIs too.

**Review by martijnvg — COMMENTED** [review]
Looks good, I left two comments.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/ExecuteEnrichPolicyAction.java (hunk 1 above)
Comment (hub-cap) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/ExecuteEnrichPolicyAction.java]: This is just a dummy response for now. I assumed we would change it to a more robust response once more information about the execution are bubbled up.

Comment (martijnvg) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/action/ExecuteEnrichPolicyAction.java]: 👍

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**